src/webapp/contexts/AppConfigProvider.tsx

```diff
@@ -0,0 +1,30 @@
+import React, { createContext, useContext } from "react";
+import { useAppConfig } from "../hooks/useAppConfig";
+import { PersistedConfig } from "../../data/entities/PersistedConfig";
+import { LogoInfo } from "../hooks/useAppConfig";
+import { CustomText } from "../../domain/entities/CustomText";
+
+interface ConfigContextState {
+    appConfig: PersistedConfig;
+    save: (settings: PersistedConfig) => void;
+    hasSettingsAccess: boolean;
+    logoInfo: LogoInfo;
+    appCustomText: CustomText;
+    hasLoaded: boolean;
+}
+
+const ConfigContext = createContext<ConfigContextState | null>(null);
+
+export const AppConfigProvider: React.FC<{ children: React.ReactNode }> = ({ children }) => {
```

Comment:
@tokland not sure about this, but we have named the `AppContext` `app-context.ts`, shall we name this file `app-config-context.ts`?